vshard/router/init.lua

```diff
@@ -1137,7 +1177,12 @@ end
 local function router_map_callrw(router, func, args, opts)
     local replicasets_to_map, err, err_id, map, rid
     local timeout, do_return_raw, bucket_ids, plain_bucket_ids, grouped_args
+    local mode = MAP_CALLRW_FULL
     if opts then
+        mode, err = router_set_map_callrw_mode(opts.mode, opts.bucket_ids)
+        if err then
+            return nil, err, nil
+        end
         timeout = opts.timeout or consts.CALL_TIMEOUT_MIN
         do_return_raw = opts.return_raw
```

Comment:
Here you assign `mode` and then overwrite it when `opts` are provided. Lets not do an overwrite. And instead simply declare the empty mode variable, and with given `opts` do what you do now, and in the `else` branch assign it to the "full" default. Then there won't be an overwrite, which in turn might save us a tiny bit of cpu.